lib/Navigation/Builder.php

```diff
@@ -429,7 +429,7 @@ protected function buildNextLevel($parentDocument, $isRoot = false, $pageCallbac
                 $page->setRelation($child->getProperty('navigation_relation'));
                 $page->setDocument($child);
 
-                if ($child->getProperty('navigation_exclude') || !$child->getPublished()) {
+                if (trim((string)$child->getProperty('navigation_name')) === '' || $child->getProperty('navigation_exclude') || !$child->getPublished()) {
```

Comment:
```suggestion
                if (empty($child->getProperty('navigation_name')) || $child->getProperty('navigation_exclude') || !$child->getPublished()) {
```

[I believe `empty()` would work here](https://www.virendrachandak.com/techtalk/php-isset-vs-empty-vs-is_null/), no need to coerce the value to a string first.

Reply:
I always hate PHP's `empty()` treating the string value `"0"` as empty as well, so I always try to keep my conditions to a strict empty string. Although the chance someone configures the navigation label to `"0"` is pretty slim, I think it's (from a users point of view) unexpected that setting that label would hide it from the menu.

Reply:
What? I did not know about that quirk. 😠

Reply:
Yep, it's actually just [casting to boolean](https://www.php.net/manual/en/language.types.boolean.php#language.types.boolean.casting) and there it makes a bit more sense again to treat `"0"` as `false` (and `"1"` as `true`). But from a string perspective the function name `empty` is a bit unfortunate...